source/client-side-encryption/tests/legacy/fle2v2-CreateCollection.yml

```diff
@@ -101,10 +101,6 @@ tests:
             command:
               create: *encrypted_collection_name
               encryptedFields: &encrypted_fields_expectation {
-                # Expect state collections are not included in the encryptedFields sent to the server.
-                "escCollection": null,
-                "ecocCollection": null,
-                "eccCollection": null,
```

Comment:
FWIW, I also changed the PHPLIB test runner to check for omitted fields in embedded documents (https://github.com/mongodb/mongo-php-library/commit/d95e7623390b8f657755f6601859d4649612a40d) when implementing [DRIVERS-2524](https://jira.mongodb.org/browse/DRIVERS-2524).

The CSFLE test runner alludes to transactions, which in turn alludes to CRUD v1. The CRUD rules for [Evaluating Matches](https://github.com/mongodb/specifications/blob/master/source/crud/tests/README.rst#evaluating-matches) doesn't differentiate between root and embedded documents. If there's some text in the transactions or CRUD test READMEs that says otherwise, please point that out as I must have missed it.

Otherwise, I reckon that removing `null` values here means that a legacy test runner is just as likely to _not_ assert anything at all.

Reply:
The legacy Transaction test README [references the Command Monitoring Spec Tests runner](https://github.com/mongodb/specifications/blob/2a51e9827a10e61617b1c34c53377edeed0f6a7b/source/transactions/tests/README.rst?plain=1#L323-L326).

> If the test includes a list of command-started events in ``expectations``,
   compare them to the actual command-started events using the
   same logic as the Command Monitoring Spec Tests runner, plus the rules in
   the Command-Started Events instructions below.

The legacy Command Monitoring Spec test format README was removed in https://github.com/mongodb/specifications/commit/c51e7373aab211ed673061bd6fdb542f655e413d#diff-5c0a965655147652fef540421431fb48533cd30c9120331276a7776ecc5171e2L62 and included:

> The driver MUST assert the expected data is present and also MUST allow for additional data to be present as well at the top level of the command document or reply document.

> However, if there are fields in command subdocuments that are not mentioned in the YAML, then the command does *not* pass the test

Reply:
Noted. I missed the reference to the Command Monitoring spec test runner.